src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,30 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateStart = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateEnd = LocalDate.parse(dateTo, formatter);
+        StringBuilder salaryReport = new StringBuilder();
+        salaryReport.append("Report for period ").append(dateFrom).append(" - ")
+                .append(dateTo).append(System.lineSeparator());
+        for (String name : names) {
+            int salaryTotal = 0;
+            for (String record : data) {
+                String[] partOfRecord = record.split(" ");
```

Comment:
Consider adding validation to ensure that `record.split(" ")` results in an array of the expected length. This will help avoid potential `ArrayIndexOutOfBoundsException` if the input data is malformed.